File: API/Entities/Office.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

[Table("Offices")]
public class Office
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string City { get; set; }
    public required string Street { get; set; }
    public required string Apartment { get; set; }
    public List<int> MondayHours { get; set; } = [];
    public List<int> TuesdayHours { get; set; } = [];
    public List<int> WednesdayHours { get; set; } = [];
    public List<int> ThursdayHours { get; set; } = [];
    public List<int> FridayHours { get; set; } = [];
    public List<int> SaturdayHours { get; set; } = [];
    public List<int> SundayHours { get; set; } = [];

    //Office - Owner(Doctor)
    public int DoctorId { get; set; }
    public AppUser Doctor { get; set; } = null!;

    //Office - Appointment
    public ICollection<Appointment> Appointments { get; set; } = [];

    //Office - Result
    public ICollection<Result> Results { get; set; } = [];
}

File: client/src/app/_models/office.ts
import { Member } from "./member";

export interface Office{
    id: number;
    name: string;
    city: string;
    street: string;
    apartment: string;
    mondayHours: number[];
    tuesdayHours: number[];
    wednesdayHours: number[];
    thursdayHours: number[];
    fridayHours: number[];
    saturdayHours: number[];
    sundayHours: number[];
    createdAt: Date;
    doctor: Member;
}
